docs/source/examples/sentiment.rst

```diff
@@ -0,0 +1,359 @@
+calculation result.
+
+When we parallelize the above operation, it is necessary to traverse nodes and perform operation A and operation
+B precisely, because the tree structure is different for each sample. However, by using the stack,
+we can calculate different tree structures by simple repeating processing. Therefore, parallelization is possible.
+
+.. literalinclude:: ../../../examples/sentiment/thin_stack.py
+   :language: python
+   :pyobject: ThinStackSet
+   :caption: thin_stack.py
+
+Let the sentence length be :math:`I`, and the number of dimensions of the hidden vector be :math:`D`,
+the thin stack can efficiently use the memory by using the matrix of :math:`(2I-1) \times D`.
+In a normal stack, we need :math:`O(I^2 D)` space computation, whereas thin stacks require :math:`O(ID)`.
+This is realized by push operations ``thin_stack_set`` and pop operations ``thin_stack_get``.
+First of all, we define ``ThinStackSet`` and ``ThinStackGet`` which inherit ``chainer.Function``.
+``ThinStackSet`` is literally a function to set values on the thin stack.
+``inputs`` in ``forward`` and `` backward``  can be broken down like ``stack, indices, values = inputs``.
+``stack`` is shared by functions by setting it as a function argument in the thin stack itself.
+Because ``chainer.Function`` does not have internal states inside, it handles ``stack`` externally by passing
+it as a function argument.
+
+.. literalinclude:: ../../../examples/sentiment/thin_stack.py
+   :language: python
+   :pyobject: ThinStackGet
+   :caption: thin_stack.py
+
+``ThinStackGet`` is literally a function to retrieve values from the thin stack.
+``inputs`` in ``forward`` and ``backward`` can be broken down like ``stack, indices = inputs``.
+
+.. literalinclude:: ../../../examples/sentiment/train_recursive_minibatch.py
+   :language: python
+   :pyobject: ThinStackRecursiveNet
+   :caption: train_recursive_minibatch.py
+
+.. literalinclude:: ../../../examples/sentiment/train_recursive_minibatch.py
+   :language: python
+   :start-after: test_data, args.batchsize, repeat=False, shuffle=False)
+   :end-before: updater = training.StandardUpdater
+   :caption: train_recursive_minibatch.py
+   :dedent: 4
+
+3.3 Preparation of Updater/Trainer and execution of training
+----------------------------------------------------------------
+
+Let's train with the new model ``ThinStackRecursiveNet``. Since we can now compute mini batches in parallel, we can see that training is faster.
+
+.. literalinclude:: ../../../examples/sentiment/train_recursive_minibatch.py
+   :language: python
+   :start-after: optimizer.setup(model) 
+   :end-before: __main__
+   :caption: train_recursive_minibatch.py
+   :dedent: 4
+
+It should get much faster!
+
+4. Reference
+=============
+* [1] `Socher, Richard; et al. "Recursive Deep Models for Semantic Compositionality Over a Sentiment Treebank". <https://nlp.stanford.edu/~socherr/EMNLP2013_RNTN.pdf>`_
+* [2] 深層学習による自然言語処理 (機械学習プロフェッショナルシリーズ) [Japanese]
```

Comment:
I think it's better to avoid reference to Japanese books here.
(You can leave this in Japanese version of colab notebook.)

Reply:
Okay!